quarkus-integration/quarkus/deployment/src/main/java/ai/timefold/solver/quarkus/deployment/TimefoldProcessor.java

```diff
@@ -225,29 +218,330 @@ SolverConfigBuildItem recordAndRegisterBeans(TimefoldRecorder recorder, Recorder
+                    declaringClass = annotationTarget.asMethod().declaringClass();
+                    break;
+                default:
+                    throw new IllegalStateException(
+                            "Member annotation @" + annotationInstance.name().withoutPackagePrefix() + " is on ("
+                                    + annotationTarget +
+                                    "), which is an invalid target type (" + annotationTarget.kind() +
+                                    ") for @" + annotationInstance.name().withoutPackagePrefix() + ".");
+            }
+
+            if (!declaringClass.annotationsMap().containsKey(DotNames.PLANNING_ENTITY)) {
+                throw new IllegalStateException(prefix + "with a @" +
+                        annotationInstance.name().withoutPackagePrefix() +
+                        " annotation is in a class (" + declaringClass.name()
+                        + ") that does not have a @" + PlanningEntity.class.getSimpleName() +
+                        " annotation.\n" +
+                        "Maybe add a @" + PlanningEntity.class.getSimpleName() +
+                        " annotation on the class (" + declaringClass.name() + ").");
+            }
+        }
+    }
+
+    private void assertSolverConfigSolutionClasses(IndexView indexView, Map<String, SolverConfig> solverConfigMap) {
+        // Validate the solution class        
+        // No solution class
+        assertEmptyInstances(indexView, DotNames.PLANNING_SOLUTION);
+        // Multiple classes and single solver
+        Collection<AnnotationInstance> annotationInstances = indexView.getAnnotations(DotNames.PLANNING_SOLUTION);
+        if (annotationInstances.size() > 1 && solverConfigMap.size() == 1) {
+            throw new IllegalStateException("Multiple classes (%s) found in the classpath with a @%s annotation.".formatted(
+                    convertAnnotationInstancesToString(annotationInstances), PlanningSolution.class.getSimpleName()));
+        }
+        // Multiple classes and at least one solver config does not specify the solution class
+        List<String> solverConfigWithoutSolutionClassList = solverConfigMap.entrySet().stream()
+                .filter(e -> e.getValue().getSolutionClass() == null)
+                .map(Map.Entry::getKey)
+                .toList();
+        if (annotationInstances.size() > 1 && !solverConfigWithoutSolutionClassList.isEmpty()) {
+            throw new IllegalStateException(
+                    """
+                            Some solver configs (%s) don't specify a %s class, yet there are multiple available (%s) on the classpath.
+                            Maybe set the XML config file to the related solver configs, or add the missing solution classes to the XML files,
+                            or remove the unnecessary solution classes from the classpath.
+                            """
```

Comment:
```suggestion
                            or remove the unnecessary solution classes from the classpath."""
```

Be mindful that, if the `"""` is on a new line, the exception message will include an empty line. You don't want that.